api/keystore/service.go

```diff
@@ -301,6 +301,60 @@ func (ks *Keystore) ImportUser(r *http.Request, args *ImportUserArgs, reply *Imp
 	return nil
 }
 
+// DeleteUserArgs are arguments for passing into DeleteUser requests
+type DeleteUserArgs struct {
+	Username string `json:"username"`
+	Password string `json:"password"`
+}
+
+// DeleteUserReply is the response from calling DeleteUser
+type DeleteUserReply struct {
+	Success bool `json:"success"`
+}
+
+// DeleteUser deletes user with the provided username and password.
+func (ks *Keystore) DeleteUser(_ *http.Request, args *DeleteUserArgs, reply *DeleteUserReply) error {
+	ks.lock.Lock()
+	defer ks.lock.Unlock()
+
+	ks.log.Verbo("DeleteUser called with %s", args.Username)
+
+	if args.Username == "" {
+		return errEmptyUsername
+	}
+
+	// check if user exists and valid user.
+	usr, err := ks.getUser(args.Username)
+	switch {
+	case err != nil || usr == nil:
+		return fmt.Errorf("user doesn't exists: %s", args.Username)
+	case !usr.CheckPassword(args.Password):
+		return fmt.Errorf("incorrect password for user %q", args.Username)
+	}
+
+	userNameBytes := []byte(args.Username)
+	userBatch := ks.userDB.NewBatch()
+	if err := userBatch.Delete(userNameBytes); err != nil {
+		return err
+	}
+
+	userDataDB := prefixdb.New(userNameBytes, ks.bcDB)
+	dataBatch := userDataDB.NewBatch()
+	if err := dataBatch.Delete(userNameBytes); err != nil {
+		return err
+	}
```

Comment:
The `userNameBytes` key might not be part of this DB, we want to delete all the users data here. So we should be iterating over this prefixed DB and deleting all the entries. So something like https://github.com/ava-labs/gecko/blob/master/api/keystore/service.go#L230-L240 but with Deleting the keys as we iterator over them.